src/layouts/RedirectIfNotLoggedIn.tsx

```diff
@@ -0,0 +1,40 @@
+import { useRouter } from "next/router";
+import React, { Fragment } from "react";
+import { Button, Modal, Text } from "../components/atomic";
+import { useAuth } from "../utils/firebase/auth";
+
+interface RedirectIfNotLoggedInProps {
+  pathAfterLoggingIn?: string;
+}
+
+const RedirectIfNotLoggedIn: React.FC<RedirectIfNotLoggedInProps> = ({
+  children,
+  pathAfterLoggingIn,
+}) => {
+  const { user } = useAuth();
+  const router = useRouter();
+  return (
+    <Fragment>
+      {children}
+      <Modal isOpen={!user} onClose={() => {}}>
```

Comment:
Would we just use the same function as the button for this onClose?

Reply:
Yeah--though this modal won't be closable so I'm passing in an empty function here.